docs/UserGuide.md

```diff
@@ -168,12 +206,169 @@ The keyword refers to the status, role or company that the user intends to look
 Examples:
 `find Google` searches for all application with `COMPANY_NAME` and/or `JOB_TITLE` as Google
 
-### Clearing all entries : `clear`
+### Clear all internship application entries : `clear`
 
-Clears all entries from the internship tracker.
+Clears all internship application entries from the internship tracker
 
 Format: `clear`
 
+### Revert recent deleted an internship application : `revert`
+
+Reverts recent delete command and restores the relevant data to the end of the current internship applications list
+
+Format: `revert`
+
+Examples:
+1. Assume the most recent delete command was `delete 2` which has data `n/Tech j/Job`, the data was removed from the applications list.
+2. Command `revert` restores the entries at the back the application list, which has effect similar to `add  n/Tech j/Job`.
+**This command is only able to restore current session's data, all the deleted / cleared data will be permanently deleted if command `exit` is executed.**
+
+### Revert recent deleted or cleared internship applications : `revert_all`
+
+Reverts all recent deleted command or cleared command and restores the affected data back to the end of the current internship applications list
+
+Format: `revert_all`
+
+**This command is only able to restore current session's data, all the deleted / cleared data will be permanently deleted if command `exit` is executed.**
+
+### Side features: Planning to apply internships 
+
+### Display lists of tasks (todos and notes) : `list_task`
+
+Displays the todo list and the note list together
+
+Format: `list`
+
+Examples:
+
+* `list_task` shows all the todos and notes that the user has written for together in one window.
+* If there are no todo and note at the moment, `No task (todo and note) at the moment` will be shown.
+* If there is either no todo or note, it will show `No todo at the moment` or `No note at the moment` respectively. The other list will be displayed normally.
+
+### Search for a task (todo and notes) : `find_task`
+
+Searches the recorded lists of todos and notes by keyword (company name in todos and note content in notes)
+
+Format: `find_task KEYWORD`
+
+Searches for the todos or notes with the specified `KEYWORD`.
+The keyword refers to the company name in todos ot the note content in notes that the user intends to look for.
+
+Examples:
+`find_task test` searches for all todos with `COMPANY_NAME` or all notes with `NOTE_CONTENT` as test.
+
+### Display a list of todo internship applications :`list_todo`
```

Comment:
I think it's better to have a space after ":" and before `list_todo` to make the whole document more unified